src/io/ipc/read/stream.rs

```diff
@@ -191,7 +208,15 @@ pub fn read_next<R: Read>(
     }
 }
 
-/// Arrow Stream reader
+/// Arrow Stream reader.
+///
+/// Once a stream is created the recommended way to interact with it is by
+/// iterating over its data. Each such iteration yields an [`Option<StreamState>`](StreamState),
+/// signaling that the stream might not currently have any data, even though it's still a "live"
+/// stream.
```

Comment:
```suggestion
/// An [`Iterator`] over an Arrow stream that yields a result of [`StreamState`]s.
/// This is the recommended way to read an arrow stream (by iterating over its data).
```

(no need to repeat what `StreamState` state is; the iterator's items are usually referred to as the thing inside the `Option`)